### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by khanh-devos**
### Tasks in this activity:
- Config a new store, create BookReducer and CategoryReducer.
- Add the new store to the app.
- Create a new folder Redux to contain the two new reducers.
- Create two actions 'addBook' and 'removeBook' inside BookReducer.
- Apply useSelector and useDispatch to modify AddNewBook and BookItem.

### CODE REVIEW

File: src/redux/categories/categorySlice.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+const initialState = {
+  categories: [],
+  amount: 0,
+  isLoading: true,
+};
+
+const categorySlice = createSlice({
+  name: 'category',
+  initialState,
+  reducers: {},
```
Comment (Ibizugbe) [inline src/redux/categories/categorySlice.js:12]: Nice job so far😉. Just as required for this milestone, we ought to create a reducer that checks the status and always returns "Under construction" 😉. Kindly do well to implement this in the reducers👍🏼.

image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @khanh-devos 👋🏼👋🏼👋🏼, 

require_changes


Good job so far!👍🏼
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] The files structure follows the duck pattern👍🏼.
- [x] @reduxjs/toolkit & react-redux are present in package.json file👍🏼.
- [x] A book slice has been created with an initiall state and a reducer that performs the add and remove actions👍🏼.
- [x] The configure store has a reducer that holds the book and categories reducer👍🏼.



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OyePriscilla — APPROVED** [review]
Hi @khanh-devos 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉
 


## Highlights
`categoryReducer` status updated ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by khanh-devos** [comment]
Hi @OyePriscilla 
Thank you for your quick review.

**Pull request merged**